metapackages/auto-instrumentations-node/src/utils.ts

```diff
@@ -136,3 +161,63 @@ export function getNodeAutoInstrumentations(
 
   return instrumentations;
 }
+
+export function getResourceDetectorsFromEnv(): Array<Detector | DetectorSync> {
+  const resourceDetectorsFromEnv =
+    process.env.OTEL_NODE_RESOURCE_DETECTORS?.split(',');
+
+  if (
+    resourceDetectorsFromEnv == null ||
+    resourceDetectorsFromEnv.includes('all')
+  ) {
+    return [
+      // Standard resource detectors.
+      containerDetector,
+      envDetectorSync,
+      hostDetectorSync,
+      osDetectorSync,
+      processDetectorSync,
+      // Cloud resource detectors.
+      alibabaCloudEcsDetector,
+      // Ordered AWS Resource Detectors as per:
+      // https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/resourcedetectionprocessor/README.md#ordering
+      awsEksDetector,
+      awsEc2Detector,
+      gcpDetector,
+    ];
+  }
+
+  const resourceDetectors: Array<Detector | DetectorSync> = [];
+  for (const detector of resourceDetectorsFromEnv) {
+    switch (detector) {
+      case RESOURCE_DETECTOR_CONTAINER:
+        resourceDetectors.push(containerDetector);
+        break;
+      case RESOURCE_DETECTOR_ENVIRONMENT:
+        resourceDetectors.push(envDetectorSync);
+        break;
+      case RESOURCE_DETECTOR_HOST:
+        resourceDetectors.push(hostDetectorSync);
+        break;
+      case RESOURCE_DETECTOR_OS:
+        resourceDetectors.push(osDetectorSync);
+        break;
+      case RESOURCE_DETECTOR_PROCESS:
+        resourceDetectors.push(processDetectorSync);
+        break;
+      case RESOURCE_DETECTOR_ALIBABA:
+        resourceDetectors.push(alibabaCloudEcsDetector);
+        break;
+      case RESOURCE_DETECTOR_AWS:
+        resourceDetectors.push(awsEksDetector);
+        resourceDetectors.push(awsEc2Detector);
+        break;
+      case RESOURCE_DETECTOR_GCP:
+        resourceDetectors.push(gcpDetector);
+        break;
+      default:
```

Comment:
The user requested for a detector which we cannot setup. I think we should `diag.error` in this case to let user know his configuration is faulty and not honored.

Reply:
Added an error message to notify the user of bad configuration :)